crates/common/src/instruction.rs

```diff
@@ -16,6 +16,8 @@ pub enum InstructionError {
     InvalidOpcode(M31),
     #[error("Size mismatch for instruction: expected {expected}, found {found}")]
     SizeMismatch { expected: usize, found: usize },
+    #[error("Assertion failed: {0} != {1}")]
+    AssertionFailed(M31, M31),
```

Comment:
should more info be introduced for improved debugging?
Maybe pc and opcode id?

Reply:
merging it this way for now; in the future, we might be interested in adding a mapping from PC to high-level line of code (via the AST), maybe with a function call stack, for even better debugging